src/components/geolocate-control.js

```diff
@@ -0,0 +1,285 @@
+/* global window */
+import {createElement, createRef} from 'react';
+import PropTypes from 'prop-types';
+import WebMercatorViewport from 'viewport-mercator-project';
+import {GeolocateControl as MapboxGeolocateControl, LngLat} from '../utils/mapboxgl';
+
+import BaseControl from './base-control';
+import Marker from './marker';
+import MapState from '../utils/map-state';
+import TransitionManager from '../utils/transition-manager';
+
+const LINEAR_TRANSITION_PROPS = Object.assign(
+  {},
+  TransitionManager.defaultProps,
+  {
+    transitionDuration: 500
+  }
+);
+
+// a copy of mapbox Event class
+class Event {
+  constructor(type, data) {
+    for (const prop in data) {
+      this[prop] = data[prop];
+    }
+
+    this.type = type;
+  }
+}
+
+const propTypes = Object.assign({}, BaseControl.propTypes, {
+  // Custom className
+  className: PropTypes.string,
+
+  // mapbox geolocate options
+  // https://docs.mapbox.com/mapbox-gl-js/api/#geolocatecontrol
+  positionOptions: PropTypes.object,
+  fitBoundsOptions: PropTypes.object,
+  trackUserLocation: PropTypes.bool,
+  showUserLocation: PropTypes.bool,
+
+  // Callbacks fired when the user interacted with the map. The object passed to the callbacks
+  // contains viewport properties such as `longitude`, `latitude`, `zoom` etc.
+  onViewStateChange: PropTypes.func,
+  onViewportChange: PropTypes.func
+});
+
+const defaultProps = Object.assign({}, BaseControl.defaultProps, {
+  className: '',
+  // mapbox geolocate options
+  positionOptions: {
```

Comment:
Is this something we have to copy, or can we use mapbox's default by not passing it in?